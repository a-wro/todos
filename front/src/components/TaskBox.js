import React from 'react'
import TaskForm from './forms/taskForm'
import { connect } from 'react-redux'
import { getTodos } from '../actions/getTodos'

export class TaskBox extends React.Component {

    componentDidMount() {
        this.props.getTodos()
    }

    render() {
        const { todos } = this.props
        return (
            <div>
                <TaskForm/>
                <div>
                  {todos.tasks.map(todo => {
                      return (
                        <div key={todo.id}>
                            {todo.task}
                        </div>
                      )
                  })}
                </div>
            </div>
        )
    }
}

const mapStateToProps = state => {
    return {
        todos: state.todos
    }
}

export default connect(mapStateToProps, { getTodos })(TaskBox)