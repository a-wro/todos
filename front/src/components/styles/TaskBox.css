
.add-button {
    text-align: center;
    padding-top: 1.25rem;
}

.todo-paper {
    margin: auto;
    width: 60%;
    padding-top: 1.25rem;
    padding-bottom: 1.25rem;
    height: 100%;
    margin-top: 0.5rem;
}

.task-box {
    margin-top: 2.5rem;
}