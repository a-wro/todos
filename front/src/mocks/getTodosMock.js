import { getTodos } from "../actions/getTodos";

//curl http:127.0.0.1:8000/todos/
export const getTodosMock =  [
    {
       "id": 1,
       "task": "test",
       "added": "2018-07-14",
       "completed": false
    },
    {
       "id": 2,
       "task": "test 2",
       "added": "2018-07-14",
       "completed": false
    }
 ]

 export default getTodosMock
