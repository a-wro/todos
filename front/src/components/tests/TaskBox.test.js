import React from 'react';
import { TaskBox } from '../TaskBox'
import TaskForm from '../forms/taskForm'
import { shallow } from 'enzyme'
import getTodosMock from '../../mocks/getTodosMock'

describe('TaskBox', () => {
  const props = { todos: { tasks: getTodosMock } }
  const taskbox = shallow(<TaskBox {...props} />)
  
  describe('Rendering', () => {
    it('renders without crashing', () => {
        expect(taskbox).toMatchSnapshot()
      })
      
      it('correctly renders a `TaskForm` component', () => {
        expect(taskbox.find(TaskForm).exists()).toBe(true)
        
      })


  })
 

})
