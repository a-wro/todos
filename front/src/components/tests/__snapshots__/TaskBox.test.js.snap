// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskBox Rendering renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TaskBox
    todos={
        Object {
            "tasks": Array [
              Object {
                "added": "2018-07-14",
                "completed": false,
                "id": 1,
                "task": "test",
              },
              Object {
                "added": "2018-07-14",
                "completed": false,
                "id": 2,
                "task": "test 2",
              },
            ],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <TaskForm />,
        <div>
          <div>
                    test
          </div>
          <div>
                    test 2
          </div>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              test
</div>,
            <div>
              test 2
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": "test",
            },
            "ref": null,
            "rendered": "test",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": "test 2",
            },
            "ref": null,
            "rendered": "test 2",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TaskForm />,
          <div>
            <div>
                        test
            </div>
            <div>
                        test 2
            </div>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                test
</div>,
              <div>
                test 2
</div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": "test",
              },
              "ref": null,
              "rendered": "test",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": "test 2",
              },
              "ref": null,
              "rendered": "test 2",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
